capstone/src/test/java/com/google/sps/servlets/ExploreServletTest.java

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalURLFetchServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.collect.ImmutableList;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+/**
+ * Class to test ExploreServlet by adding a few mock clubs and ensuring their existence and
+ * properties
+ */
+@RunWith(JUnit4.class)
+public final class ExploreServletTest {
+
+  private static final String KEVIN = "kshao@google.com";
+  private static final String MEGHA = "kakm@google.com";
+  private static final String MEGAN = "meganshi@google.com";
+
+  private static final String CLUB_1 = "Club 1";
+  private static final String CLUB_2 = "Club 2";
+  private static final String CLUB_3 = "Club 3";
+
+  private static final String SITE_1 = "www.fakesite.com";
+  private static final String SITE_2 = "www.realfakesite.com";
+
+  private static final String DESCRIPTION_1 = "Helping people.";
+  private static final String DESCRIPTION_2 = "Not helping people.";
```

Comment:
nit: I don't think these need to be constants. They can be hard-coded directly into the tests which is actually more readable/preferable in tests because the whole scope of the test is contained inside it.